src/libappimage/CMakeLists.txt

```diff
@@ -44,24 +42,22 @@ foreach(target libappimage libappimage_static)
         target_link_libraries(${target} PRIVATE -static-libgcc -static-libstdc++)
     endif()
 
-    target_include_directories(${target} PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>)
+    set_property(TARGET ${target} PROPERTY PUBLIC_HEADER ${libappimage_public_header})
 
-    set_property(TARGET libappimage PROPERTY PUBLIC_HEADER ${libappimage_public_header})
+    set_property(TARGET ${target} PROPERTY VERSION ${libappimage_VERSION})
+    set_property(TARGET ${target} PROPERTY SOVERSION ${libappimage_SOVERSION})
 
-    set_property(TARGET libappimage PROPERTY VERSION ${libappimage_VERSION})
-    set_property(TARGET libappimage PROPERTY SOVERSION ${libappimage_SOVERSION})
+    # install libappimage
+    install(
```

Comment:
I somewhere read that using `PUBLIC_HEADER` like this is actually wrong. Is there any reason you use that property? Are there alternative ways? (IIRC, you introduced that years ago, and I was always curious.)

Reply:
You are right, using `PUBLIC_HEADER` in Unix system is wrong, that property only has a real effect on MacOS.